physionet-django/search/views.py

```diff
@@ -53,3 +52,88 @@ def all_topics(request):
     topics = PublishedTopic.objects.all().order_by('-project_count')
 
     return render(request, 'search/all_topics.html', {'topics':topics})
+
+
+def get_content(resource_type, orderby, direction):
+    """
+    Helper function to get content shown on a resource listing page
+    """
+    if resource_type is None:
+        published_projects = PublishedProject.objects.all()
+    else:
+        published_projects = PublishedProject.objects.filter(
+            resource_type=resource_type)
+
+    direction = '-' if direction == 'desc' else ''
+
+    order_string = '{}{}'.format(direction, orderby)
+    published_projects = published_projects.order_by(order_string)
+
+    authors = [p.authors.all() for p in published_projects]
+    topics = [p.topics.all() for p in published_projects]
+    projects_authors_topics = zip(published_projects, authors, topics)
+
+    return projects_authors_topics
+
+
+def content_index(request):
+    """
+    List of all published resources
+    """
+    orderby, direction = 'publish_datetime', 'desc'
+    form = forms.ProjectOrderForm()
+
+    if 'orderby' in request.GET or 'direction' in request.GET:
+        form = forms.ProjectOrderForm(request.GET)
+        if form.is_valid():
+            orderby, direction = [form.cleaned_data[item] for item in ['orderby', 'direction']]
+        projects_authors_topics = get_content(resource_type=None,
+            orderby=orderby, direction=direction)
+    else:
+        projects_authors_topics = get_content(resource_type=None,
+            orderby=orderby, direction=direction)
+
+    return render(request, 'search/content_index.html', {'form':form,
+        'projects_authors_topics':projects_authors_topics})
+
+
+def database_index(request):
```

Comment:
Why have separate views for `database_index` and `software_index`? Seems like it would be easier to handle the type with an argument passed from urls.py, particularly as we are likely to be adding new content types in the near future.

Reply:
I don't have a strong opinion about this, but Roger wanted that list of data categories at the top (which I we haven't implemented) yet so that would be a rather large difference. For now, there isn't too much boiler code, and is quite clear without needing to pass in a lot of string variables just for display.

Reply:
A single approach for all projects (single template/function) is more future proof. The current approach may seem fine now, but if someone decides to modify each of the functions/templates independently then it becomes tricky to try to merge them back into a single function/template.